frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -886,7 +1092,31 @@ mod tests {
 			let encoded = pool.read().transactions[0].clone();
 			let extrinsic: Extrinsic = Decode::decode(&mut &*encoded).unwrap();
 			let call = extrinsic.call;
-			assert!(matches!(call, OuterCall::MultiPhase(Call::submit_unsigned(_, _))));
+			assert!(matches!(call, OuterCall::MultiPhase(Call::submit_unsigned(..))));
+		})
+	}
+
+	#[test]
+	fn ocw_solution_must_have_correct_round() {
+		let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
+		ext.execute_with(|| {
+			roll_to_with_ocw(25);
+			assert_eq!(MultiPhase::current_phase(), Phase::Unsigned((true, 25)));
+			// OCW must have submitted now
+			// now, before we check the call, update the round
+			<crate::Round<Runtime>>::mutate(|round| *round += 1);
+
+			let encoded = pool.read().transactions[0].clone();
+			let extrinsic = Extrinsic::decode(&mut &*encoded).unwrap();
+			let solution = match extrinsic.call {
+				OuterCall::MultiPhase(Call::submit_unsigned(solution, _)) => solution,
+				_ => panic!("bad call: unexpected submission"),
+			};
+
+			assert_noop!(
+				MultiPhase::unsigned_pre_dispatch_checks(&solution),
```

Comment:
Correct semantic is to call `validate_unsigned` and `pre_dispatch`, if you want to ensure that something is rejected by the pool. 

This is a helper function that _may or may not_ be called in the same code path, and this test is assuming that it is, which is incorrect.

Reply:
Ok, thanks for pointing that out. 9c2804878